compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -487,7 +492,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             sym::bitreverse => (bits << extra).reverse_bits(),
             _ => bug!("not a numeric intrinsic: {}", name),
         };
-        Ok(Scalar::from_uint(bits_out, layout.size))
+        Ok(Scalar::from_uint(bits_out, ret_layout.size))
```

Comment:
This looks like it can ICE when bswap/bitrevere is called with a too small ret_layout.

Reply:
How would that happen though? The intrinsics are defined like `pub fn bitreverse<T: Copy>(x: T) -> T;`, so the `ret_layout` must be the same?

Reply:
That's an extremely non-local invariant. This function here should make sense on its own without worrying about what happens in the standard library.

Reply:
Maybe add an assertion that for bscap/bitrevers, the two layouts are the same.

Reply:
I think this is true in general with intrinsics, we implicitly depend on their signature in a lot of ways. Either way, I added an assertion.

Reply:
In the interpreter and Miri we always have local assertions for these things.